### PR TITLE
[onDeviceUI] Adding on device addons.

### PR DESCRIPTION
**Pull request by Gongreg**
This PR is part of #3903.

This adds 3 on device addons: Notes, Knobs, Backgrounds.
They are definitely not the cleanest and up to date addons.

Looking forward @ndelangen suggested that we have to think of the way to display addons as webviews in the app. But for now these are good enough to display that we can have addons in the app.

### CODE REVIEW

File: addons/ondevice-backgrounds/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@storybook/addon-ondevice-backgrounds",
+  "version": "4.0.0-alpha.20",
```
Comment (igor-dv) [inline addons/ondevice-backgrounds/package.json]: missed the version update

File: addons/ondevice-backgrounds/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "url": "https://github.com/storybooks/storybook.git"
+  },
+  "license": "MIT",
+  "main": "dist/index.js",
+  "jsnext:main": "src/index.js",
+  "scripts": {
+    "prepare": "node ../../scripts/prepare.js"
+  },
+  "dependencies": {
+    "@storybook/addons": "4.0.0-alpha.20",
```
Comment (igor-dv) [inline addons/ondevice-backgrounds/package.json]: like everywhere

File: addons/ondevice-knobs/package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "license": "MIT",
+  "main": "dist/index.js",
+  "jsnext:main": "src/index.js",
+  "scripts": {
+    "prepare": "node ../../scripts/prepare.js"
+  },
+  "dependencies": {
+    "@storybook/addons": "4.0.0-alpha.20",
+    "@storybook/addon-knobs": "4.0.0-alpha.20",
```
Comment (igor-dv) [inline addons/ondevice-knobs/package.json]: didn't get where this in use

File: addons/ondevice-knobs/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "deep-equal": "^1.0.1",
+    "prop-types": "^15.6.2",
+    "react-native-color-picker": "^0.4.0",
+    "react-native-modal-datetime-picker": "^5.1.0",
+    "react-native-modal-selector": "0.0.27",
```
Comment (igor-dv) [inline addons/ondevice-knobs/package.json]: should this be stuck to the exact version?

**Review by igor-dv — APPROVED** [review]
Mostly versions mismatch.

File: addons/ondevice-knobs/package.json (hunk 3 above)
Comment (Gongreg) [reply addons/ondevice-knobs/package.json]: A person who uses ondevice-knobs will be still importing addon-knobs inside their app. So if he doesn't install it directly we still want him to have it in his node_modules. @igor-dv.

Comment (igor-dv) [reply addons/ondevice-knobs/package.json]: Yes, but users will need to install it explicitly. In that case, you probably want it to be a peer

Comment (Gongreg) [reply addons/ondevice-knobs/package.json]: Alright, thanks :)

**Comment by Gongreg** [comment]
Nothing seems to be blocking this PR too @ndelangen, @igor-dv.

**Comment by igor-dv** [comment]
Let's merge if after the #3903 is ready? We won't use addons without the support for addons, right?

**Comment by Gongreg** [comment]
Sure.

**Comment by ndelangen** [comment]
merge when ready

**Review by ndelangen — APPROVED** [review]

**Pull request merged**